### PR TITLE
legendary-gl: init at 0.0.14

### PR DESCRIPTION
**Pull request by wchresta**
###### Motivation for this change

* Adds requested launcher for EPIC games
* Was requested in #88963

###### Things done

I have started up the binary and authenticated with an empty account. Because I don't own any EPIC games, I couldn't test any of the `real` functionality, but login works, at least.

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbogdan** [comment]
Thank you for the PR! So far I tried

- authenticating
- listing the games under my account
- installing a game

All worked no problem - I'm now just waiting for a download to complete. Due to the quality of my current internet connection it might take a while but will report back as soon as that's complete.

I'm not however familiar with Python packaging so can't really offer any comments on the expression itself.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23755,6 +23755,8 @@ in
 
   leela-zero = libsForQt5.callPackage ../games/leela-zero { };
 
+  legendary-gl = callPackage ../games/legendary-gl { };
```
Comment (prusnak) [inline pkgs/top-level/all-packages.nix:23758]:
```suggestion
  legendary-gl = python3Packages.callPackage ../games/legendary-gl { };
```

File: pkgs/games/legendary-gl/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, fetchFromGitHub
+, python38Packages
```
Comment (prusnak) [inline pkgs/games/legendary-gl/default.nix:3]:
```suggestion
, buildPythonApplication
```

File: pkgs/games/legendary-gl/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+}:
+
+with python38Packages;
+
```
Comment (prusnak) [inline pkgs/games/legendary-gl/default.nix:7]: Remove this
```suggestion
```

File: pkgs/games/legendary-gl/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "05r88qi8mmbj07wxcpb3fhbl40qscbq1aqb0mnj9bpmi9gf5zll5";
+  };
+
+  propagatedBuildInputs = [
+    requests
+  ];
+
+  meta = with lib; {
+    description = "A free and open-source Epic Games Launcher alternative";
+    homepage = https://github.com/derrod/legendary;
```
Comment (prusnak) [inline pkgs/games/legendary-gl/default.nix:25]: Use quotes for URLs (RFC45)
```suggestion
    homepage = "https://github.com/derrod/legendary";
```

File: pkgs/games/legendary-gl/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    license = licenses.gpl3;
+    maintainers = [ maintainers.wchresta ];
```
Comment (prusnak) [inline pkgs/games/legendary-gl/default.nix:27]:
```suggestion
    maintainers = with maintainers; [ wchresta ];
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/games/legendary-gl/default.nix (hunk 3 above)
Comment (wchresta) [reply pkgs/games/legendary-gl/default.nix:7]: I added this because the app requires Py >= 3.8. What's a better way?

Comment (wchresta) [reply pkgs/games/legendary-gl/default.nix]: Got it, using `python38Packages.callPackage` in all-packages :)

File: pkgs/games/legendary-gl/default.nix (hunk 5 above)
Comment (wchresta) [reply pkgs/games/legendary-gl/default.nix]: Thanks :)

File: pkgs/games/legendary-gl/default.nix (hunk 4 above)
Comment (wchresta) [reply pkgs/games/legendary-gl/default.nix]: Thanks :)

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (wchresta) [reply pkgs/top-level/all-packages.nix]: Thanks :)

File: pkgs/games/legendary-gl/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "derrod";
+    repo = "legendary";
+    rev = version;
+    sha256 = "05r88qi8mmbj07wxcpb3fhbl40qscbq1aqb0mnj9bpmi9gf5zll5";
+  };
+
+  propagatedBuildInputs = [ requests ];
+
```
Comment (prusnak) [inline pkgs/games/legendary-gl/default.nix:20]:
```suggestion
  disabled = pythonOlder "3.8";
```

File: pkgs/games/legendary-gl/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{ lib
+, fetchFromGitHub
+, buildPythonApplication
+, requests
```
Comment (prusnak) [inline pkgs/games/legendary-gl/default.nix:4]: Add `pythonOlder` here

File: pkgs/games/legendary-gl/default.nix [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+}:
+
+# Requires python >= 3.8
```
Comment (prusnak) [inline pkgs/games/legendary-gl/default.nix:7]: you can drop this comment once you add `disabled = ...` below

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/games/legendary-gl/default.nix (hunk 8 above)
Comment (wchresta) [reply pkgs/games/legendary-gl/default.nix]: Done :)

**Review by prusnak — APPROVED** [review]

**Comment by pbogdan** [comment]
Sorry for the late response - tested this so far with a single game and all steps worked fine though launching required providing a shell that included `wine`. I'm not quite sure if it would make sense to create a wrapper with `wine` as user may want to use different version depending on the game. Other than this LGTM.

**Review by pbogdan — APPROVED** [review]

**Pull request merged**